tests/system.rs

```diff
@@ -1720,6 +1725,39 @@ fn test_clang_cache_whitespace_normalization(
     }
 }
 
+// Checks that the -include$PWD/test.h flag does not get re-ordered and
+// placed somewhere other than directly after the -Xarch_<arch> flag.
+// If it gets re-ordered, the compilation will fail.
+fn test_clang_xarch_flags(compiler: Compiler, tempdir: &Path) {
```

Comment:
I added an e2e test here, based on the original issue comment about reordered `-include` flag breaking compilation.